nncf/common/graph/graph.py

```diff
@@ -33,51 +33,73 @@ class NNCFNode:
     Class describing nodes used in NNCFGraph.
     """
 
-    def __init__(self, node_id: int, node_name: NNCFNodeName, data: dict = None):
-        self.node_id = node_id
-        self.data = data if data else {}
-        self.data[NNCFGraph.NODE_NAME_ATTR] = node_name
+    ID_NODE_ATTR = "id"
```

Comment:
May be add dataclass for layer attribute instead of use names?
```python
@dataclass
class NodeAttributes:
    id: int
    name: str
    type: str
    ....
```

Or more, it's really need to save node attribute in `attribute` may be save node attribute in the class attributes?
```python
@dataclass
class NNCFNode:
    id: int
    name: str
    type: str
    ....
    data # for extra data like `output_mask`
```

Reply:
This is necessary to avoid copying attributes from the networkx representation to NNCFNode. This is main optimization in this PR.